### PR TITLE
[Train][Doc]  Improve AIR Lightning APIs and docstrings

### PR DESCRIPTION
**Pull request by woshiyyya**
## Why are these changes needed?

During the LightningTrainer bug-bash session, we collected a lot of constructive feedback to improve the user experience. This PR aims to address these feedbacks:

**LightningTrainer**:
- Updated docstring: remove "All other ModelCheckpoint callbacks will be ignored."
- Allow users to provide their customized ProgressBar callback.

**LightningCheckpoint**:
- Added examples for `.from_directory()` and `.get_model()` usage in the docstring.
- Changed `load_from_checkpoint_kwargs` argument type from dict to **kwargs.
- Added an error message in `checkpoint.from_path()`: if the provided path is a directory, then instruct them to provide a file path, or to use from_directory() instead.

**LightningConfigBuilder**:
- Add docstrings in `checkpointing()`, `fit_params()`, `module()`, and `trainer()`
- Raise a warning if the user provides "model" in `.fit_params()`.
- Tell people to still provide an AIR `CheckpointConfig` even if they have already set it in the `LightningConfigBuilder.checkpointing()`.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [x] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/train/lightning/lightning_trainer.py [hunk 1]
```diff
@@ -131,8 +148,19 @@ def ddp_strategy(self, **kwargs) -> "LightningConfigBuilder":
     def checkpointing(self, **kwargs) -> "LightningConfigBuilder":
         """Set up the configurations of ``pytorch_lightning.callbacks.ModelCheckpoint``.
 
-        LightningTrainer creates a `ModelCheckpoint` callback based on this config.
-        The AIR checkpointing and logging methods are triggered in that callback.
+        LightningTrainer creates a subclass instance of the `ModelCheckpoint` callback
+        with the kwargs. It handles checkpointing and metrics logging logics.
+
+        Specifically, the callback periodically reports the latest metrics and
+        checkpoint to the AIR session via `session.report()`. The report frequency
```
Comment (krfricke) [inline python/ray/train/lightning/lightning_trainer.py:155]:
```suggestion
        checkpoint to the AIR session via ``session.report()``. The report frequency
```
nit

Comment (krfricke) [reply python/ray/train/lightning/lightning_trainer.py:155]: Actually, maybe use `:ref:`?

File: python/ray/train/lightning/lightning_trainer.py [hunk 2]
```diff
@@ -131,8 +148,19 @@ def ddp_strategy(self, **kwargs) -> "LightningConfigBuilder":
...
+        matches the checkpointing frequency here. You have to make sure that the
+        target metrics (e.g. metrics defined in `TuneConfig` or `CheckpointConfig`)
```
Comment (krfricke) [inline python/ray/train/lightning/lightning_trainer.py:157]: Same here, ideally use `ref` to reference the API docs

**Review by krfricke — APPROVED** [review]
Looks good to me so far!

File: python/ray/train/lightning/lightning_trainer.py (hunk 1 above)
Comment (woshiyyya) [reply python/ray/train/lightning/lightning_trainer.py]: Done! Actually I found we can link it to api doc with 
```
:meth:`session.report() <ray.air.session.report>` 
```

File: python/ray/train/lightning/lightning_trainer.py (hunk 2 above)
Comment (woshiyyya) [reply python/ray/train/lightning/lightning_trainer.py]: Done!

File: python/ray/train/lightning/lightning_trainer.py [hunk 3]
```diff
@@ -110,16 +115,28 @@ def trainer(self, **kwargs) -> "LightningConfigBuilder":
     def fit_params(self, **kwargs) -> "LightningConfigBuilder":
         """The parameter lists for ``pytorch_lightning.Trainer.fit()``
 
+        ``LightningTrainer`` creates a model instance with the parameters provided
+        in `.module()` and feeds it into the ``pl.Trainer.fit()`` method.
```
[the remark this answers is not in the document]
Comment (woshiyyya) [reply python/ray/train/lightning/lightning_trainer.py:119]: Oh actually, sphinx can magically link it to the method api if it's under the same scope.

File: python/ray/train/lightning/lightning_checkpoint.py [hunk 4]
```diff
@@ -38,25 +58,25 @@ def from_path(
...
-            ...     path="/path/to/checkpoint.ckpt"
-            ... )
         """
 
         assert os.path.exists(path), f"Lightning checkpoint {path} doesn't exists!"
 
+        if os.path.isdir(path):
+            raise ValueError(
+                f"`from_path()` expects a file path, but `{path}` is a directory. "
+                "Please provide a valid checkpoint file path, or try to use "
```
[the remark this answers is not in the document]
Comment (woshiyyya) [reply python/ray/train/lightning/lightning_checkpoint.py]: Done.

**Review by matthewdeng — APPROVED** [review]

**Review by richardliaw — APPROVED** [review]

**Review by zhe-thoughts — APPROVED** [review]
Approved for merging to master. Low risk docs / examples change, in a new component

**Pull request merged**